codemash/codemash/CodeMashUtilities.swift

```diff
@@ -121,10 +122,53 @@ func favoriteSession(id: Int, isFavorited: Bool) {
     } else {
         favorites = favorites.filter({ $0 != id })
     }
-
+    
     prefs.set(favorites, forKey: favKey)
 }
 
+func setSelectedDay(day: Day) {
+    prefs.set(day.rawValue, forKey: selectedSessionDayKey)
+}
+
+func getSelectedDay() -> Day {
+    //if saved day is less than today's date go to today
+    let savedDow = getDayFromInt(dowInt: prefs.integer(forKey: selectedSessionDayKey))
+    let currentDow: Int = getAdjustedDow(dow: getCurrentDayOfWeek())
+    let currentDowAdjusted: Day = getDayFromInt(dowInt: currentDow)
+    
+    if(savedDow.rawValue < currentDow){
+        return currentDowAdjusted
+    }else{
+        return savedDow
+    }
+}
+
+func getCurrentDayOfWeek() -> Int {
+    let dateForm = DateFormatter()
+    dateForm.dateFormat = "e"
+    if let dow = Int(dateForm.string(from: Date())) {
+        return dow
+    }else{
+        return 1
+    }
+}
+
+func getAdjustedDow(dow: Int) -> Int{
+    if dow >= 3{
+        return dow - 2
+    }else{
+        return 1
+    }
+}
+
+func getDayFromInt(dowInt: Int) -> Day {
+    if let day = Day(rawValue: dowInt) {
+        return day
+    }else{
+        return Day.Tuesday
+    }
+}
+
```

Comment:
instead of putting the returns in an else block, then can live outside it, since you're returning in the if.
getSelectedDay, getCurrentDayOfWeek, getAdjustedDow, getDayFromInt

func getDayFromInt(dowInt: Int) -> Day {
     if let day = Day(rawValue: dowInt) {
         return day
     }
     return Day.Tuesday
 }

Reply:
Got it!